raiden-ts/src/channels/epics.ts

```diff
@@ -878,3 +890,56 @@ export const channelSettleableEpic = (
       }
     }),
   );
+
+/**
+ * Process new blocks and re-emit confirmed or removed actions
+ *
+ * @param action$ - Observable of channelSettle actions
+ * @param state$ - Observable of RaidenStates
+ * @param deps - RaidenEpicDeps
+ * @param deps.config$,deps.provider - RaidenEpicDeps members
+ * @returns Observable of confirmed or removed actions
+ */
+export const confirmationEpic = (
+  {}: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { config$, provider }: RaidenEpicDeps,
+): Observable<ConfirmableAction> =>
+  combineLatest(
+    state$.pipe(pluckDistinct('blockNumber')),
+    state$.pipe(pluck('pendingTxs')),
+    config$.pipe(pluckDistinct('confirmationBlocks')),
+  ).pipe(
+    // exhaust will ignore blocks while concat$ is busy
+    exhaustMap(([blockNumber, pendingTxs, confirmationBlocks]) =>
+      concat$(
+        ...pendingTxs
+          // only txs/confirmable actions which are more than confirmationBlocks in the past
+          .filter(a => a.payload.txBlock + confirmationBlocks <= blockNumber)
+          .map(a =>
+            defer(() => provider.getTransactionReceipt(a.payload.txHash)).pipe(
+              map(receipt => {
+                if (
```

Comment:
nitpick: any chance we could move the map a to function to avoid nesting?

Reply:
Good catch. Will do, thanks!